src/guard.c

```diff
@@ -44,16 +47,35 @@ unguard_pages(tsdn_t *tsdn, ehooks_t *ehooks, edata_t *edata, emap_t *emap) {
 	uintptr_t guard1 = addr - PAGE;
 	uintptr_t guard2 = addr + size;
 
-	assert(edata_state_get(edata) == extent_state_active);
+	assert(edata_state_get(edata) == extent_state_active
+	    || edata_state_get(edata) == extent_state_retained);
```

Comment:
nit: Split into 2 assertions in the first `if (reg_emap)` branch? I assume the retained only holds if !reg_emap

Reply:
Right, let's do that.